### ISSUE TITLE
No stdout/stderr from run_process

### ISSUE THREAD

**Issue by hairmare** [savonet/liquidsoap#437]
I just got around to really trying out liquidsoap 1.3.0, sorry for being late to the party. I've had way to much support to do over the last couple of weeks.

## Issue
`run_process` does not return any output with a git pinned opam liquidsoap 1.3.0 on Ubuntu Trusty from opam and Centos 7.3 from my [packages](https://build.opensuse.org/project/show/home:radiorabe:liquidsoap). 

### stdout is empty
When I run this:

```bash
liquidsoap 'print(run_process("date"))'
```
#### Actual Output
```
(("",""),("exit","0"))
```

#### Expected Output

I would assume that the `fst(fst())` pair should contain the stdout from date as described by `liquidsoap -h run_process`, ie. something similar to the following.
```
(("Sat Apr 29 07:45:24 UTC 2017",""),("exit","0"))
```

### stderr is empty

The same thing is happening with stderr

```
liquidsoap 'print(run_process("echo nope 1>&2"))'`
```

#### Actual Output
```
(("",""),("exit","0"))
```

#### Expected Output

```
(("","nope"),("exit","0"))
```

## Additional Info
The log output shows nothing conclusive if I do a `--verbose --debug --force-start`.

```
(("",""),("exit","0"))
2017/04/29 08:32:02 >>> LOG START
2017/04/29 08:32:02 [main:3] Liquidsoap 1.3.0
2017/04/29 08:32:02 [main:3] Using: graphics=[distributed with Ocaml] bytes=1.1 pcre=6.2.5 dtools=0.3.3 duppy=0.6.0 cry=0.4.1 mm=0.3.0 xmlplaylist=0.1.4 ogg=0.5.1 vorbis=0.6.2 opus=0.1.1 speex=0.2.1 mad=0.4.5 flac=0.1.2 flac.ogg=0.1.2 dynlink=[distributed with Ocaml] lame=0.3.3 fdkaac=0.2.1 theora=0.3.1 alsa=0.2.2 samplerate=0.1.3 taglib=0.3.3 ssl=0.5.2 magic=0.7.3 inotify=2.3 yojson=1.1.8 faad=0.3.3 soundtouch=0.1.8 ladspa=0.1.5
2017/04/29 08:32:02 [video.converter:4] Couldn't find preferred video converter: gavl.
2017/04/29 08:32:02 [audio.converter:4] Using preferred samplerate converter: libsamplerate.
2017/04/29 08:32:02 [threads:3] Created thread "generic queue #1".
2017/04/29 08:32:02 [threads:3] Created thread "generic queue #2".
2017/04/29 08:32:02 [threads:3] Created thread "non-blocking queue #1".
2017/04/29 08:32:02 [threads:3] Created thread "non-blocking queue #2".
2017/04/29 08:32:02 [clock:4] Main phase starts.

^C

2017/04/29 08:32:04 [main:3] Shutdown started!
2017/04/29 08:32:04 [main:3] Waiting for threads to terminate...
2017/04/29 08:32:04 [main:3] Threads terminated.
2017/04/29 08:32:04 [threads:3] Shutting down scheduler...
2017/04/29 08:32:04 [threads:3] Scheduler shut down.
2017/04/29 08:32:04 [main:3] Cleaning downloaded files...
2017/04/29 08:32:04 [threads:3] Thread "generic queue #1" terminated.
2017/04/29 08:32:04 [main:3] Freeing memory...
2017/04/29 08:32:04 >>> LOG END
```

Please let me know if you think trying another distro like Debian would help narrow the issue down.

**Comment by geoffroymontel** [comment]
Hello

I have a similar problem.
I'm on Ubuntu 14.04TLS and the following code does not work

```
lines = get_process_lines("echo hi")
log("lines = #{lines}")
line = list.hd(default="",lines)
log("line = #{line}")
```

returns 

```
2017/05/05 15:24:42 [lang:3] lines = []
2017/05/05 15:24:42 [lang:3] line = 
```

Thanks in advance

**Comment by toots** [comment]
This is a bug in the OCaml compiler. See: https://caml.inria.fr/mantis/view.php?id=7136

I'll commit a fix in the code and close this issue. I don't think we need to re-releasing, the issue has been fixed in OCaml 4.03 and above so I would recommend to upgrade your compiler for the time being.

Thanks for reporting.

**Comment by toots** [comment]
Actually, looking at default compiler versions in Debian and Ubuntu, I think that we're gonna have to release a bugfix version soon, unfortunately..

**Comment by hairmare** [comment]
Thanks for the feedback. I got it working locally and was just looking into the package situation for CentOS. It's ocaml is even [more outdated (4.01)](https://centos.pkgs.org/7/centos-x86_64/ocaml-4.01.0-22.7.el7_2.x86_64.rpm.html) than what is [in debian (4.03)](https://tracker.debian.org/pkg/ocaml). On CentOS ocaml is based on a package from upstream RHEL so getting it updated by upstream could be an issue as such a change would need considerable support in RHELs customer base.

I'm backporting the fix from a9ef199e9ac9087f932cfd1c379bfa7fe543f341 into [RaBe's CentOS packages](https://build.opensuse.org/project/show/home:radiorabe:liquidsoap) but like you said, that won't help with opam installs.

**Comment by toots** [comment]
Thanks! opam package can be patched but it's an important issue so I'd rather have it fixed in a new release..
